numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -621,6 +594,31 @@ static PyObject *
 /**end repeat**/
 
 static PyObject *
+voidtype_str(PyObject *self){
+    PyObject *arr, *ret = NULL;
+
+    if (!PyDataType_HASFIELDS(((PyVoidScalarObject*)self)->descr)) {
+        PyObject  *item, *item_str;
+
+        item = gentype_generic_method(self, NULL, NULL, "item");
+        if (item == NULL) {
+            return NULL;
+        }
+
+        item_str = PyObject_Str(item);
+        Py_DECREF(item);
+        return item_str;
+    }
```

Comment:
This returns a different result to `1.12` - previously, it gave `(1., 2.)`, now it gives `(1.0, 2.0)`.

Perhaps `void.item()` should not convert fields to python scalars?